include/bm/bm_sim/source_info.h

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2017 Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Andy Fingerhut (jafinger@cisco.com)
+ *
+ */
+
+#ifndef BM_BM_SIM_SOURCE_INFO_H_
+#define BM_BM_SIM_SOURCE_INFO_H_
+
+//! @file source_info.h
+//! Stores source location information about some objects, as read from
+//! the JSON file produced by the compiler.
+
+#include <string>
+#include <sstream>
+
+namespace bm {
+
+class SourceInfo {
+ public:
+  SourceInfo()
+    : filename(""), line(0), column(0), source_fragment("") { }
+  SourceInfo(std::string filename, unsigned int line, unsigned int column,
+             std::string source_fragment)
+    : filename(filename),
+      line(line),
+      column(column),
+      source_fragment(source_fragment) { }
+
+  std::string toString() const {
```

Comment:
my recommendation is to pre-compute that string in the constructor and store it as a class member as this is not such a cheap operation
also could this happen in an implementation file, to avoid the dependency on `sstream` in the header file